02-microarray/pathway-analysis_microarray_03_gsva.Rmd

```diff
@@ -405,11 +424,222 @@ Note that if we had duplicate gene identifiers here, we would not be able to set
 
 ## GSVA - Microarray
 
-_Addressed in upcoming PR_
+Rather than ranking genes based on some gene-level statistic selected by the user, GSVA fits a model and ranks genes based on their expression level relative to the sample distribution.
+The pathway-level score calculated is a way of asking how genes _within_ a gene set vary as compared to genes that are _outside_ of that gene set [@Malhotra2018].
+
+The idea here is that we will get pathway-level scores for each sample that indicate if genes in a pathway vary concordantly in one direction (over-expressed or under-expressed relative to the overall population) [@Hanzelmann2013].
+
+The output is a gene set by sample matrix of GSVA scores.
+
+### Perform GSVA
+
+Let's perform GSVA using the `gsva()` function.
+
+```{r}
+gsva_results <- gsva(
+  filtered_mapped_matrix,
+  hallmarks_list,
+  method = "gsva",
+  # Appropriate for our log2-transformed microarray data
+  kcdf = "Gaussian",
+  # Minimum gene set size
+  min.sz = 15,
+  # Maximum gene set size
+  max.sz = 500,
+  # Compute Gaussian-distributed scores
+  mx.diff = TRUE,
+  # Don't print out the progress bar
+  verbose = FALSE
+)
+```
+
+Note that the `gsva()` function documentation says we can use `kcdf = "Gaussian"` if we have expression values that are continuous such as log-CPMs, log-RPKMs or log-TPMs, but we would use `kcdf = "Poisson"` on integer counts.
+See `?gsva` for more options.
+
+```{r}
+# Let's explore what the output of gsva() looks like
+head(gsva_results[, 1:10])
+```
+
+## Find the pathway with the most significant differences across treatment groups
+
+If we want to know which pathway is most differentially expressed across subgroups, we can pretty easily use GSVA scores with differential expression steps using the limma package. 
+
+### Create the design matrix
```

Comment:
For reviewing purposes, I'll note that most of this is a fairly direct grab from https://alexslemonade.github.io/refinebio-examples/02-microarray/differential-expression_microarray_02_several-groups.html#44_Create_the_design_matrix and has been reviewed before. 

Only difference is the subgroups are slightly different.